osu.Game/Screens/Multi/RealtimeMultiplayer/RealtimeRoomManager.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Logging;
+using osu.Game.Online.Multiplayer;
+using osu.Game.Online.RealtimeMultiplayer;
+using osu.Game.Screens.Multi.Components;
+
+namespace osu.Game.Screens.Multi.RealtimeMultiplayer
+{
+    public class RealtimeRoomManager : RoomManager
+    {
+        [Resolved]
+        private StatefulMultiplayerClient multiplayerClient { get; set; }
+
+        public readonly Bindable<double> TimeBetweenListingPolls = new Bindable<double>();
+        public readonly Bindable<double> TimeBetweenSelectionPolls = new Bindable<double>();
+        private readonly IBindable<bool> isConnected = new Bindable<bool>();
+        private readonly Bindable<bool> allowPolling = new Bindable<bool>();
+
+        private ListingPollingComponent listingPollingComponent;
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            isConnected.BindTo(multiplayerClient.IsConnected);
+            isConnected.BindValueChanged(_ => Schedule(updatePolling), true);
+
+            updatePolling();
+        }
+
+        public override void CreateRoom(Room room, Action<Room> onSuccess = null, Action<string> onError = null)
+            => base.CreateRoom(room, r => joinMultiplayerRoom(r, onSuccess), onError);
+
+        public override void JoinRoom(Room room, Action<Room> onSuccess = null, Action<string> onError = null)
+            => base.JoinRoom(room, r => joinMultiplayerRoom(r, onSuccess), onError);
+
+        public override void PartRoom()
+        {
+            if (JoinedRoom == null)
+                return;
+
+            var joinedRoom = JoinedRoom;
+
+            base.PartRoom();
+            multiplayerClient.LeaveRoom().Wait();
```

Comment:
You'll have to ignore the fact it's not async and returning completed tasks. Eventually there'll be an override of this that is async, which is why the method returns a `Task`. Are you suggesting to fire-and-forget? I don't really like that as it potentially means someone could join another room after "parting" but then still receive events from the previous room (if network latency is high to the multiplayer server). I'd like to avoid such a possibility for now.

Reply:
No, fire and forget is miles worse. It's just a habit of mine to question every `.Result`/`.Wait()` call as more often than not they're trouble down the line.

Reply:
I guess the main issue I have with this is because it's not explicitly task returning and not mentioned in xmldoc, it's a bit hidden in the `PartRoom` call. In this case it's probably fine because there won't be many instances of it being called, but it's feasible that in a future we call `PartRoom` somewhere that shouldn't be blocking (or worse, could be in threadpool already and cause a deadlock).

I think adding something to the xmldoc of the `IRoomManager` interface may be good? Just a simple mention that "this call blocks on the operation being sent to the server" (for `JoinRoom` too I guess?).

Also of note is that in the case of signalr the blockingness is only until the command has been queued for sending; it's not blocking on an actual network operation (for now).

Reply:
I've just changed it to fire-and-forget for now and do additional `Room == null` checks. Server will throw errors if trying to join while not parted anyway.